src/biokbase/narrative/jobs/job.py

```diff
@@ -341,16 +364,18 @@ def _update_state(self, state: dict) -> None:
         # Check if there would be no change in updating
         # i.e., if state <= self._acc_state
         if self._acc_state is not None:
-            if {**self._acc_state, **state} == self._acc_state:
+            if merge(self._acc_state, state) == self._acc_state:
                 return
 
         state = copy.deepcopy(state)
         if self._acc_state is None:
             self._acc_state = state
         else:
-            self._acc_state.update(state)
+            merge_inplace(self._acc_state, state)
+
+        self.last_updated = time.time_ns()
 
-    def state(self, force_refresh=False, exclude=JOB_INIT_EXCLUDED_JOB_STATE_FIELDS):
+    def refresh_state(self, force_refresh=False, exclude=JOB_INIT_EXCLUDED_JOB_STATE_FIELDS):
```

Comment:
I would think that `refresh_state` would only be called if `force_refresh` is true -- otherwise the caller would use `static_state`

Reply:
Maybe you're right. It's ancient logic though and maybe in the scope of another, less urgent PR?

Reply:
Maybe call it `latest_state` or something ...

Reply:
Looks like `force_refresh` is just for getting `child_jobs` and `retry_ids`